sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_client.py

```diff
@@ -401,25 +431,40 @@ def list_anomalies_for_detection_configuration(self, detection_configuration_id,
         :raises ~azure.core.exceptions.HttpResponseError:
         """
 
-        skip = kwargs.pop('skip', None)
-        filter_condition = kwargs.pop('filter', None)
-        converted_start_time = convert_datetime(start_time)
-        converted_end_time = convert_datetime(end_time)
-        detection_anomaly_result_query = DetectionAnomalyResultQuery(
-            start_time=converted_start_time,
-            end_time=converted_end_time,
-            filter=filter_condition,
-        )
-
-        return self._client.get_anomalies_by_anomaly_detection_configuration(  # type: ignore
-            configuration_id=detection_configuration_id,
-            skip=skip,
-            body=detection_anomaly_result_query,
-            cls=lambda objs: [DataPointAnomaly._from_generated(x) for x in objs],
-            **kwargs)
+    @distributed_trace
+    def list_anomalies(self, **kwargs):
```

Comment:
maybe this is more pythonic and I'm being over-critical, but in intellisense it's going to be tough to tell which are the subset of required parameters needed without documentation or some trial and error. I'm wondering if it makes sense to have one required parameter called something like `filter_by: Union[DetectionConfigQuery, AlertQuery]` that contains the required params for each type of query. This could also be used by list_incidents.

Reply:
actually looks quite good in intellisense 👍 

![image](https://user-images.githubusercontent.com/31998003/97612243-70bb0d00-19d4-11eb-8ea2-7a70ca6b120a.png)